ruby/tests/basic.rb

```diff
@@ -724,9 +756,12 @@ def test_dup
       m = TestMessage.new
       m.optional_string = "hello"
       m.optional_int32 = 42
-      m.repeated_msg.push TestMessage2.new(:foo => 100)
-      m.repeated_msg.push TestMessage2.new(:foo => 200)
-
+      tm1 = TestMessage2.new(:foo => 100)
+      tm2 = TestMessage2.new(:foo => 200)
+      m.repeated_msg.push tm1
+      assert m.repeated_msg[-1] == tm1
+      m.repeated_msg.push tm2
+      assert m.repeated_msg[-1] == tm2
```

Comment:
Ah, I think what you're saying is that *with your change to behavior*, this test no longer passes. And I see why -- it actually exposed a lower-level bug in the JRuby implementation. Thanks!